lib/spack/spack/spec.py

```diff
@@ -527,6 +527,13 @@ def __init__(self, spec_like, *dep_like, **kwargs):
             # XXX(deptype): default deptypes
             self._add_dependency(spec, ('build', 'link'))
 
+    def __getattr__(self, item):
+        """Delegate to self.package if the attribute is not in the spec"""
+        # This line is to avoid infinite recursion in case package is
+        # not present among self attributes
+        package = super(Spec, self).__getattribute__('package')
+        return getattr(package, item)
```

Comment:
@adamjstewart Try to substitute the current function with:
```python
def __getattr__(self, item):                                                                                                                                                                                   
        """Delegate to self.package if the attribute is not in the spec"""                                                                                                                                         
        # This line is to avoid infinite recursion in case package is                                                                                                                                              
        # not present among self attributes                                                                                                                                                                        
        package = self.__dict__['package']                                                                                                                                                                         
        return getattr(package, item)
```
I am trying it right now but that should work...